linode/vpcips/framework_models.go

```diff
@@ -0,0 +1,78 @@
+	Address      types.String `tfsdk:"address"`
+	AddressRange types.String `tfsdk:"address_range"`
+	Gateway      types.String `tfsdk:"gateway"`
+	SubnetMask   types.String `tfsdk:"subnet_mask"`
+	Prefix       types.Int64  `tfsdk:"prefix"`
+	LinodeID     types.Int64  `tfsdk:"linode_id"`
+	Region       types.String `tfsdk:"region"`
+	Active       types.Bool   `tfsdk:"active"`
+	NAT1To1      types.String `tfsdk:"nat_1_1"`
+	VPCID        types.Int64  `tfsdk:"vpc_id"`
+	SubnetID     types.Int64  `tfsdk:"subnet_id"`
+	ConfigID     types.Int64  `tfsdk:"config_id"`
+	InterfaceID  types.Int64  `tfsdk:"interface_id"`
+}
+
+func (m *VPCIPModel) FlattenVPCIP(vpcIp *linodego.VPCIP, preserveKnown bool) {
+	m.Address = helper.KeepOrUpdateStringPointer(m.Address, vpcIp.Address, preserveKnown)
+	m.AddressRange = helper.KeepOrUpdateStringPointer(m.AddressRange, vpcIp.AddressRange, preserveKnown)
+	m.Gateway = helper.KeepOrUpdateString(m.Gateway, vpcIp.Gateway, preserveKnown)
+	m.SubnetMask = helper.KeepOrUpdateString(m.SubnetMask, vpcIp.SubnetMask, preserveKnown)
+	m.Prefix = helper.KeepOrUpdateInt64(m.Prefix, int64(vpcIp.Prefix), preserveKnown)
+	m.LinodeID = helper.KeepOrUpdateInt64(m.LinodeID, int64(vpcIp.LinodeID), preserveKnown)
+	m.Region = helper.KeepOrUpdateString(m.Region, vpcIp.Region, preserveKnown)
+	m.Active = helper.KeepOrUpdateBool(m.Active, vpcIp.Active, preserveKnown)
+	m.NAT1To1 = helper.KeepOrUpdateStringPointer(m.NAT1To1, vpcIp.NAT1To1, preserveKnown)
+	m.VPCID = helper.KeepOrUpdateInt64(m.VPCID, int64(vpcIp.VPCID), preserveKnown)
+	m.SubnetID = helper.KeepOrUpdateInt64(m.SubnetID, int64(vpcIp.SubnetID), preserveKnown)
+	m.ConfigID = helper.KeepOrUpdateInt64(m.ConfigID, int64(vpcIp.ConfigID), preserveKnown)
+	m.InterfaceID = helper.KeepOrUpdateInt64(m.InterfaceID, int64(vpcIp.InterfaceID), preserveKnown)
+}
+
+func (m *VPCIPModel) CopyFrom(other VPCIPModel, preserveKnown bool) {
+	m.Address = helper.KeepOrUpdateValue(m.Address, other.Address, preserveKnown)
+	m.AddressRange = helper.KeepOrUpdateValue(m.AddressRange, other.AddressRange, preserveKnown)
+	m.Gateway = helper.KeepOrUpdateValue(m.Gateway, other.Gateway, preserveKnown)
+	m.SubnetMask = helper.KeepOrUpdateValue(m.SubnetMask, other.SubnetMask, preserveKnown)
+	m.Prefix = helper.KeepOrUpdateValue(m.Prefix, other.Prefix, preserveKnown)
+	m.LinodeID = helper.KeepOrUpdateValue(m.LinodeID, other.LinodeID, preserveKnown)
+	m.Region = helper.KeepOrUpdateValue(m.Region, other.Region, preserveKnown)
+	m.Active = helper.KeepOrUpdateValue(m.Active, other.Active, preserveKnown)
+	m.NAT1To1 = helper.KeepOrUpdateValue(m.NAT1To1, other.NAT1To1, preserveKnown)
+	m.VPCID = helper.KeepOrUpdateValue(m.VPCID, other.VPCID, preserveKnown)
+	m.SubnetID = helper.KeepOrUpdateValue(m.SubnetID, other.SubnetID, preserveKnown)
+	m.ConfigID = helper.KeepOrUpdateValue(m.ConfigID, other.ConfigID, preserveKnown)
+	m.InterfaceID = helper.KeepOrUpdateValue(m.InterfaceID, other.InterfaceID, preserveKnown)
+}
+
```

Comment:
`CopyFrom` is actually not needed for a data source, because it's only required to eliminate remaining unknown values after flatten of an linodego object after an update API call in the `Update` function in a resource

Reply:
Thanks for explaining! I removed it 👍